lib/selinux/selinux_linux.go

```diff
@@ -0,0 +1,145 @@
+ */
+
+package selinux
+
+import (
+	"bufio"
+	"bytes"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gravitational/trace"
+	ocselinux "github.com/opencontainers/selinux/go-selinux"
+)
+
+const (
+	selinuxConfig        = "/etc/selinux/config"
+	selinuxTypeTag       = "SELINUXTYPE"
+	permissiveModuleName = "permissive_" + domain
+)
+
+// copied from github.com/opencontainers/selinux/go-selinux/selinux-linux.go
+func readConfig(target string) string {
+	in, err := os.Open(selinuxConfig)
+	if err != nil {
+		return ""
+	}
+	defer in.Close()
+
+	scanner := bufio.NewScanner(in)
+
+	for scanner.Scan() {
+		line := bytes.TrimSpace(scanner.Bytes())
+		if len(line) == 0 {
+			// Skip blank lines
+			continue
+		}
+		if line[0] == ';' || line[0] == '#' {
+			// Skip comments
+			continue
+		}
+		fields := bytes.SplitN(line, []byte{'='}, 2)
+		if len(fields) != 2 {
+			continue
+		}
+		if bytes.Equal(fields[0], []byte(target)) {
+			return string(bytes.Trim(fields[1], `"`))
+		}
+	}
+	return ""
+}
+
+// CheckConfiguration returns an error if SELinux is not configured to
+// enforce the SSH service correctly.
+func CheckConfiguration() error {
+	if !ocselinux.GetEnabled() {
+		return trace.Errorf("SELinux is disabled or not present")
+	}
+	if ocselinux.EnforceMode() != ocselinux.Enforcing {
+		return trace.Errorf("SELinux mode is not enforcing, SELinux will not constrain anything")
+	}
+
+	selinuxType := readConfig(selinuxTypeTag)
+	if selinuxType == "" {
+		return trace.NotFound("could not find SELinux type")
+	}
+	selinuxDir := filepath.Join("/var/lib/selinux", selinuxType, "active/modules")
+
+	var moduleInstalled, moduleDisabled, modulePermissive bool
+	err := filepath.WalkDir(selinuxDir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		name := d.Name()
+		if strings.Contains(name, moduleName) {
+			moduleInstalled = true
+			if name == permissiveModuleName {
+				modulePermissive = true
+				// if the module is permissive, we also know it's installed
+				// so we can end the walk
+				return filepath.SkipAll
+			}
+			if filepath.Base(filepath.Dir(path)) == "disabled" {
+				moduleDisabled = true
+				// if the module is disabled, we also know it's installed
+				// so we can end the walk
+				return filepath.SkipAll
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		return trace.Wrap(err, "failed to find SSH SELinux module")
+	}
+
+	if !moduleInstalled {
+		return trace.Errorf("the SSH SELinux module %s is not installed", moduleName)
+	}
+	if moduleDisabled {
+		return trace.Errorf("the SSH SELinux module %s is disabled", moduleName)
+	}
+	if modulePermissive {
+		return trace.Errorf("the SSH SELinux module %s is permissive, so policy denials will be logged but not enforced", moduleName)
+	}
+
+	return nil
+}
+
+// UserContext returns the SELinux context that should be used when
+// creating processes as a certain user.
+func UserContext(login string) (string, error) {
```

Comment:
The `login` value is an actual selinux user (e.g. `user_u`), right? Not a linux user?

Reply:
Nope, that's the name of a Linux user, an SELinux user is what's returned